src/ConnectedMode/StatefulServerBranchProvider.cs

```diff
@@ -34,15 +34,21 @@ internal sealed class StatefulServerBranchProvider : IStatefulServerBranchProvid
         private readonly IServerBranchProvider serverBranchProvider;
         private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
         private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
         private bool disposedValue;
         private string selectedBranch;
 
         [ImportingConstructor]
-        public StatefulServerBranchProvider(IServerBranchProvider serverBranchProvider, IActiveSolutionBoundTracker activeSolutionBoundTracker, ILogger logger)
+        public StatefulServerBranchProvider(
+            IServerBranchProvider serverBranchProvider,
+            IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            ILogger logger,
+            IThreadHandling threadHandling)
```

Comment:
I used the MEF import here rather than `ThreadHandling.Instance` since the connected mode project doesn't currently reference the VSInfrastructure project.